package/MDAnalysis/coordinates/DCD.py

```diff
@@ -550,106 +307,123 @@ def timeseries(self, asel=None, start=None, stop=None, step=None, skip=None,
 
         .. deprecated:: 0.16.0
            `skip` has been deprecated in favor of the standard keyword `step`.
-
         """
         if skip is not None:
             step = skip
-            warnings.warn("Skip is deprecated and will be removed in"
-                          "in 1.0. Use step instead.",
-                          category=DeprecationWarning)
+            warnings.warn(
+                "Skip is deprecated and will be removed in"
+                "in 1.0. Use step instead.",
+                category=DeprecationWarning)
 
         start, stop, step = self.check_slice_indices(start, stop, step)
 
         if asel is not None:
             if len(asel) == 0:
-                raise NoDataError("Timeseries requires at least one atom to analyze")
+                raise NoDataError(
+                    "Timeseries requires at least one atom to analyze")
             atom_numbers = list(asel.indices)
         else:
             atom_numbers = list(range(self.n_atoms))
 
-        if len(format) != 3 and format not in ['afc', 'acf', 'caf', 'cfa', 'fac', 'fca']:
-            raise ValueError("Invalid timeseries format")
-       # Check if the atom numbers can be grouped for efficiency, then we can read partial buffers
-        # from trajectory file instead of an entire timestep
-        # XXX needs to be implemented
-        return self._read_timeseries(atom_numbers, start, stop, step, format)
+        frames = self._file.readframes(
+            start, stop, step, order=format, indices=atom_numbers)
+        return frames.xyz
 
-    def close(self):
-        if self.dcdfile is not None:
-            self._finish_dcd_read()
-            self.dcdfile.close()
-            self.dcdfile = None
 
-    def Writer(self, filename, **kwargs):
-        """Returns a DCDWriter for `filename` with the same parameters as this DCD.
+class DCDWriter(base.WriterBase):
+    """DCD Writer class
 
-        Defaults for all values are obtained from the `DCDReader` itself but
-        all values can be changed through keyword arguments.
+    The writer follows recent NAMD/VMD convention for the unitcell but still
+    writes AKMA time. The unitcell will be written as ``[A, gamma, B, beta,
+    alpha, C]``
 
-        Parameters
+    """
+    format = 'DCD'
+    multiframe = True
+    flavor = 'CHARMM'
```

Comment:
Right now we don't write box vectors. I would leave it as that for the current implementation. But it sure would be easier to add now.